build/gen_stub.php

```diff
@@ -391,31 +391,33 @@ class Type {
     /** @var SimpleType[] */
     public $types;
 
-    /**
-     * @param SimpleType[] $types
-     */
-    public function __construct(array $types) {
-        $this->types = $types;
-    }
+    /** @var bool */
+    public $isReferenceSupported;
```

Comment:
Why do we need a separate flag for this? Can't we treat this as a "normal" type?

Reply:
The main reason is that I wanted to avoid treating it as a normal type because semantically it's not one. However, to be honest, doing so wouldn't cause any problems currently, since we don't use the array value type for anything besides generating optimizer info.